### DISCUSSION TITLE
Can repository admin install a GitHub App?

### DISCUSSION THREAD

**Discussion by gorohash** [community/community#50488] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

The GitHub documentation states the following

https://docs.github.com/en/apps/creating-github-apps/creating-github-apps/about-apps#about-github-apps
> To install a GitHub App, you must be an organization owner or have admin permissions in a repository.

However, I can't install a GitHub App to the repository which I am a owner.

Perhaps it means that only organization owners can install to a repository in an organization, and only repository admins can install to a repository in an personal account?

**Comment by Anmol-Baranwal** [comment]
It is generally true that in order to install a GitHub App, you must have administrative permissions on the repository or organization where you want to install the app. However, there are some additional factors that can affect whether you are able to install an app.

For example, some apps may require specific permissions or settings in order to be installed, and these requirements may vary depending on the app and the context in which it is being installed.

In addition, there may be other factors that can affect whether you are able to install an app, such as restrictions on the types of apps that can be installed in certain environments (e.g. enterprise environments).

**Reply by gorohash** [reply]
Does it mean different roles are required depending on the permissions that required by the GitHub App? Is there documentation that shows which permissions require which roles?

**Reply by Anmol-Baranwal** [reply]
Yes, the roles required to install a GitHub App may depend on the permissions required by the app. Typically, installing a GitHub App requires certain permissions or access to specific resources, such as repositories or organization settings. Depending on the level of access required, the role needed to install the app may vary.

GitHub provides documentation on the permissions required for each app and the corresponding roles needed to install the app. The specific permissions required by each app can be found in the app's documentation or in the app's listing on the GitHub Marketplace. The roles needed to install the app can be found in the GitHub documentation on installing GitHub Apps.

**Reply by Anmol-Baranwal** [reply]
@gorohash 
I just wanted to check in and see if you have any remaining questions or concerns about my answer. If not, would you please mark my response as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by gorohash** [reply]
Sorry for the late response.

Now I'm trying to install the [Renovate App](https://github.com/marketplace/renovate) to my repository in an organization. I am a admin of the repository and not a owner of the organization. I confirmed that the app requires only repository access by installing it to my personal account. However I can't choose the organization on installation page of the app. Why is that?

**Reply by Anmol-Baranwal** [reply, marked as answer]
It's possible that the organization settings are configured to restrict the installation of third-party apps to only be available to organization owners, rather than organization admins or members.

If you're not the owner of the organization, you may need to contact the organization owner and ask them to install the Renovate App for you. Alternatively, the organization owner could grant you the necessary permissions to install the app by adjusting the organization settings.

It's also possible that the Renovate App is not currently approved for use within the organization, in which case you would need to contact the organization owner or administrator to request approval.

**Reply by Anmol-Baranwal** [reply]
@gorohash 
I just wanted to check in and see if you have any remaining questions or concerns about my answer. If not, would you please mark my response as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by gorohash** [reply]
@Anmol-Baranwal 
Ultimately, I asked the owner of the organization to install the app.

My original question about the unclear description in the GitHub documentation has not been resolved. However, I achieved my goal of getting the app installed. So I marked your response as answer.

Thank you for your help.

**Comment by eleftherias** [comment]
@Anmol-Baranwal 
> It's possible that the organization settings are configured to restrict the installation of third-party apps to only be available to organization owners, rather than organization admins or members.

I haven't seen this setting anywhere. Could you point me to where I can find it?

**Reply by Anmol-Baranwal** [reply]
They may have updated the docs since GitHub always improves the settings.
You can refer to official docs if possible. I don't know about that at this current moment.
